app/javascript/controllers/search_bar_controller.js

```
@@ -201,4 +264,220 @@ export default class extends Controller {
+
+        this.activeSuggestionIndex = -1;
+        panel.classList.add('show');
+    }
+
+    /**
+     * Insert a suggestion keyword into the input, replacing the partial word
+     * that the user was typing.  Adds a trailing space for operators, keeps
+     * cursor tight for filter prefixes (e.g. "tag:").
+     */
+    insertSuggestion(keyword) {
+        const value = this.inputTarget.value;
```

ESLint

[error] Use object destructuring.

(prefer-destructuring)